src/flashbar/collapsible-flashbar.tsx

```diff
@@ -0,0 +1,392 @@
+                          (item as StackableItem).collapsedIndex ?? (item as StackableItem).expandedIndex ?? index,
+                      }
+                    : undefined
+                }
+                key={getItemId(item)}
+              >
+                {showInnerContent(item) && (
+                  <Flash
+                    // eslint-disable-next-line react/forbid-component-props
+                    className={clsx(
+                      animateFlash && styles['flash-with-motion'],
+                      isVisualRefresh && styles['flash-refresh']
+                    )}
+                    key={getItemId(item)}
+                    ref={shouldUseStandardAnimation(item, index) ? transitionRootElement : undefined}
+                    transitionState={shouldUseStandardAnimation(item, index) ? state : undefined}
+                    {...item}
+                  />
+                )}
+              </li>
+            )}
+          </Transition>
+        ))}
+      </TransitionGroup>
+    </ul>
+  );
+
+  return (
+    <div
+      {...baseProps}
+      className={clsx(
+        baseProps.className,
+        styles.flashbar,
+        styles[`breakpoint-${breakpoint}`],
+        styles.stack,
+        isFlashbarStackExpanded && styles.expanded,
+        getVisualContextClassname('flashbar')
+      )}
+      ref={mergedRef}
+    >
+      <>
+        {isFlashbarStackExpanded && renderList()}
+        {isCollapsible && (
+          <div
+            className={clsx(
+              styles.toggle,
+              isVisualRefresh && styles['visual-refresh'],
+              isFlashbarStackExpanded ? styles.expanded : styles.collapsed,
+              transitioning && styles['animation-running']
+            )}
+            onClick={toggleCollapseExpand}
+            ref={toggleElementRef}
+          >
+            <span className={styles.status} role="status">
+              {toggleButtonText && <h2 className={styles.text}>{toggleButtonText}</h2>}
+              <span className={styles['types-count']} id={itemCountElementId}>
+                <LiveRegion>
+                  {counterTypes
+                    .map(
+                      ({ type, labelName }: { type: FlashType; labelName: LabelName }) =>
+                        `${countByType[type]} ${(i18nStrings && i18nStrings[labelName]) ?? ''}`
+                    )
+                    .join(', ')}
+                </LiveRegion>
+                <span aria-hidden="true" className={styles['types-count-visible']}>
+                  {counterTypes.map(({ type, labelName, iconName }) => (
+                    <NotificationTypeCount
+                      key={type}
+                      iconName={iconName}
+                      label={i18nStrings ? i18nStrings[labelName] : undefined}
+                      count={countByType[type]}
+                    />
+                  ))}
+                </span>
+              </span>
+            </span>
+            <button
```

Comment:
Can we use `<Button variant="icon">` here?

Reply:
The arrow rotates 180 degrees instead of being a different arrow per each state, even though I know the animation is not very noticeable (especially because it happens at the same time than a more prominent animation). This was part of the UI spec.

But I based my implementation on the Button's and tried to reuse what I could, to maximize consistency, like the `default-text-style` mixin. This button is basically made of these 4 rules:

```
@include typography.default-text-style;
cursor: pointer;
margin-top: 0;
margin-bottom: 0;
```